person_class.rb

```diff
@@ -0,0 +1,28 @@
+class Person
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000),
+          @name = name,
+          @age = age,
+          @parent_permission = parent_permission
+  end
+
+  def name(value)
+    @name = value
+  end
+
+  def age(value)
+    @age = value
+  end
+
+  def can_use_services?
+    true
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
```

Comment:
*[OPTIONAL]*

- Hey, a more concise and better way to write this is can be seen below:
- Since we can now read the age and name, we don't need to explicitly access them from a parameter as you've done. so this could be further simplified below:
- These lines of code are doing the same thing in a short way making it faster to process.


```
def of_age?
    @age >= 18
  end

  def can_use_services?
    @age >= 18 || parent_permission
  end

  private  :is_of_age?
end
```

Reply:
### NAMING OF THE FILES [OPTIONAL]

- Hey, I think naming the files according to their name would make it readable as suppose to adding the `_class` to each file as you've done. e.g `person_class` to `person.rb`. etc. This is more readable. Please, kindly fix this.